### PR TITLE
[#163519143] create user profile

### PR DESCRIPTION
**Pull request by Nziranziza**
#### What does this PR do?
Add create profile 
#### Description of Task to be completed?
- create middleware 
- add controller
- add route
- add test
#### How should this be manually tested?
- git clone 
- npm install
- npm run migrate
- npm start
- use postman add visit:
```
PUT api/v1/user
```
send this as request body
```
{
  "user":{
             "firstName":
             "lastName":
             "username":
             "email":
             "image":
         }
}
```
N.B: Those fields are not mandatory and you can still add cover, bio and birthDate
#### What are the relevant pivotal tracker stories?
#163519143

### CODE REVIEW

**Comment by Nziranziza** [comment]
@oesukam @NkFab can you review may PR?

File: routes/validators/user.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        .min(20),
+      gender: Joi.string()
+        .max(6),
+      birthDate: Joi.date(),
+      image: Joi.string(),
+      cover: Joi.string(),
+    })
+  };
+
+  export default profile;
```
Comment (oesukam) [inline routes/validators/user.js]: @Nziranziza Add an empty line after line 24

File: middlewares/verifyJwt.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      if (!user) { const status = info.message === 'user does not exist'? 404:401;
+      return res.status(status).send({ errors: {
+          body: [info.message]
+      }}); }
+        req.currentUser = user;
+        return next();
+    })(req, res, next);
+}
+
+export default verifyJwt;
```
Comment (oesukam) [inline middlewares/verifyJwt.js]: @Nziranziza Add an empty line after line 17

Comment (oesukam) [reply middlewares/verifyJwt.js]: @Nziranziza you should also use two spaces instead of 4, for your tab

File: __tests__/routes/profile.test.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+          .put(`${urlPrefix}/user`)
+          .set('Authorization', token)
+          .send({user: {
+            ...profile
+          }})
+    
+        expect(res.status).toBe(520);
+        expect(res.body.errors.body).toBeDefined();
+      });
+});
```
Comment (oesukam) [inline __tests__/routes/profile.test.js]: @Nziranziza Add an empty new line after line 188

File: routes/api/user.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+
+const router = express.Router();
+
+router.put('/', verifyJwt, 
+  celebrate({ body: profileValidator }),
+  ProfileController.createProfile);
+
+
+export default router;
```
Comment (oesukam) [inline routes/api/user.js]: @Nziranziza Add a new line after line 15

**Review by oesukam — CHANGES_REQUESTED** [review]
@Nziranziza The route is working fine but you just need to add some empty lines at the end of some files that I commented on and also change your file tabulation space to 2 instead of 4

File: middlewares/verifyJwt.js (hunk 2 above)
Comment (Nziranziza) [reply middlewares/verifyJwt.js]: done!!!

File: __tests__/mocks/db.json [hunk 5]
```diff
@@ -14,5 +14,16 @@
...
+    "username": "doe201",
+    "email": "john.doe@andela.com",
+    "bio": "I am software at Andela",
+    "gender": "Male",
+    "birthDate": "12 June 1999",
+    "image": "https://planetbotanix.com/wp-content/uploads/2017/08/Female-Avatar-1-300x300.jpg",
+    "cover": "https://www.eta.co.uk/wp-content/uploads/2012/09/Cycling-by-water-resized-min.jpg"
   }
-}
+}
```
Comment (kimotho-njoki) [inline __tests__/mocks/db.json]: Empty line people. Please add an empty line at the end of each file. I do not want to see these red markers.

File: __tests__/routes/profile.test.js [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        expect(res.body.user.lastName).toBe(profile.lastName);
+        expect(res.body.user.username).toBe(profile.username);
+        expect(res.body.user.email).toBe(profile.email);
+        expect(res.body.user.bio).toBe(profile.bio);
+        expect(res.body.user.gender).toBe(profile.gender);
+        expect(res.body.user.birthDate).toBeDefined();
+        expect(res.body.user.image).toBe(profile.image);
+        expect(res.body.user.cover).toBe(profile.cover);
+      });
+      test('should create profile and not send confirmation email when email is not provided', async () => {
```
Comment (kimotho-njoki) [inline __tests__/routes/profile.test.js]: Please space out each test. For example, add an empty line after line 52.

**Review by NkFab — APPROVED** [review]

**Review by oesukam — APPROVED** [review]

File: controllers/ProfileController.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
+import uuid from 'uuid';
+import { User } from '../database/models';
+import { sendEmailConfirmationLink } from './MailController';
+/**
```
Comment (kimotho-njoki) [inline controllers/ProfileController.js:5]: empty line please after the imports

File: controllers/ProfileController.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+ * @description ProfileContoller class
+ */
+class ProfileController {
+  /**
+   * @author Daniel
+   * @param {*} req 
+   * @param {*} res
+   * @returns{*} user object
```
Comment (kimotho-njoki) [inline controllers/ProfileController.js]: space between `returns` and `{*}`

File: routes/validators/user.js (hunk 1 above)
Comment (kimotho-njoki) [inline routes/validators/user.js]: empty line after line 24!

**Pull request merged**